File: solutions/codeforces/405C.cpp
#include <stdio.h>

int A[1010][1010];

int main()
{
  int n;
  scanf("%d", &n);
  for (int i = 0; i < n; i++)
    for (int j = 0; j < n; j++)
      scanf("%d", &A[i][j]);
  int res = 0;
  for (int i = 0; i < n; i++)
    for (int j = 0; j < n; j++)
      res += A[i][j]*A[j][i];
  res %= 2;

  int k;
  scanf("%d", &k);
  while (k--) {
    int op, t;
    scanf("%d", &op);
    if (op == 3) {
      printf("%d", res);
    } else {
      res = 1-res;
      scanf("%d", &t);
    }
  }
  printf("\n");
  return 0;
}

File: solutions/codeforces/405C.py
n = int(raw_input())
A = []
for _ in xrange(n):
	A.append(map(int, raw_input().split()))

res = 0
for i in xrange(n):
	for j in xrange(n):
		res += A[i][j]*A[j][i]
res %= 2

ans = []
k = int(raw_input())
for _ in xrange(k):
	op = map(int, raw_input().split())
	if op[0] == 3:
		ans.append(res)
	else:
		res = 1-res
print ''.join(map(str, ans))
